source/Public/Install-SqlDscServer.ps1

```diff
@@ -0,0 +1,1398 @@
+        [Parameter(ParameterSetName = 'CompleteFailoverCluster')]
+        [System.Management.Automation.SwitchParameter]
+        $ASProviderMSOLAP,
+
+        [Parameter(ParameterSetName = 'InstallRole')]
+        [System.String]
+        $FarmAccount,
+
+        [Parameter(ParameterSetName = 'InstallRole')]
+        [System.Management.Automation.PSCredential]
+        $FarmPassword,
+
+        [Parameter(ParameterSetName = 'InstallRole')]
+        [System.Management.Automation.PSCredential]
+        $Passphrase,
+
+        [Parameter(ParameterSetName = 'InstallRole')]
+        [ValidateRange(0, 65536)]
+        [System.UInt16]
+        $FarmAdminiPort, # cspell: disable-line
+
+        [Parameter(ParameterSetName = 'Install')]
+        [Parameter(ParameterSetName = 'InstallRole')]
+        [Parameter(ParameterSetName = 'CompleteImage')]
+        [Parameter(ParameterSetName = 'Upgrade')]
+        [ValidateSet('Automatic', 'Disabled', 'Manual')]
+        [System.String]
+        $BrowserSvcStartupType,
+
+        [Parameter(ParameterSetName = 'Upgrade')]
+        [ValidateSet('Rebuild', 'Reset', 'Import')]
+        [System.String]
+        $FTUpgradeOption,
+
+        [Parameter(ParameterSetName = 'Install')]
+        [Parameter(ParameterSetName = 'InstallRole')]
+        [Parameter(ParameterSetName = 'CompleteImage')]
+        [System.Management.Automation.SwitchParameter]
+        $EnableRanU,
+
+        [Parameter(ParameterSetName = 'Install')]
+        [Parameter(ParameterSetName = 'InstallRole')]
+        [Parameter(ParameterSetName = 'CompleteImage')]
+        [Parameter(ParameterSetName = 'InstallFailoverCluster', Mandatory = $true)]
+        [Parameter(ParameterSetName = 'CompleteFailoverCluster', Mandatory = $true)]
+        [System.String]
+        $InstallSqlDataDir,
+
+        [Parameter(ParameterSetName = 'Install')]
+        [Parameter(ParameterSetName = 'InstallRole')]
+        [Parameter(ParameterSetName = 'CompleteImage')]
+        [Parameter(ParameterSetName = 'InstallFailoverCluster')]
+        [Parameter(ParameterSetName = 'CompleteFailoverCluster')]
+        [System.String]
+        $SqlBackupDir,
+
+        [Parameter(ParameterSetName = 'Install')]
+        [Parameter(ParameterSetName = 'InstallRole')]
+        [Parameter(ParameterSetName = 'CompleteImage')]
+        [Parameter(ParameterSetName = 'InstallFailoverCluster')]
+        [Parameter(ParameterSetName = 'CompleteFailoverCluster')]
+        [ValidateSet('SQL')]
+        [System.String]
+        $SecurityMode,
+
+        [Parameter(ParameterSetName = 'Install')]
+        [Parameter(ParameterSetName = 'InstallRole')]
+        [Parameter(ParameterSetName = 'CompleteImage')]
+        [Parameter(ParameterSetName = 'RebuildDatabase')]
+        [Parameter(ParameterSetName = 'InstallFailoverCluster')]
+        [Parameter(ParameterSetName = 'CompleteFailoverCluster')]
+        [System.Management.Automation.PSCredential]
+        $SAPwd,
```

Comment:
Shouldn't this just be a `SecureString` instead of a full PSCredential object?

Reply:
Will update this an all other password parameters to `SecureString`.

Reply:
Changed to SecureString on all password parameters.

Reply:
@wsmelton thank you for your comments. Much appreciated! 🙇